packages/tools/devtools/devtools-view/src/components/data-visualization/EditableValueView.tsx

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import { Dropdown, Input, InputOnChangeData, Option, tokens } from "@fluentui/react-components";
+import { EditType, FluidObjectValueNode, SendEditData } from "@fluid-experimental/devtools-core";
+import { useMessageRelay } from "../../MessageRelayContext";
+/**
+ * Input to {@link TreeItem}
+ */
+export interface EditableValueViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+}
+
+/**
+ * Constructs a tree element from the provided header and child contents.
+ *
+ * Intended to be used inside an outer {@link @fluentui/react-components/unstable#Tree} context.
+ */
+export function EditableValueView(props: EditableValueViewProps): React.ReactElement {
+	const { node, containerKey } = props;
+	const messageRelay = useMessageRelay();
+	const [value, setValue] = React.useState<string>(String(node.value));
+	const [isEditing, setIsEditing] = React.useState<boolean>(false);
+	const [isTextBox, setIsTextBox] = React.useState<boolean>(
+		node.typeMetadata === "number" || node.typeMetadata === "SharedCounter" ? false : true,
+	);
+	const textAreaRef = React.useRef<HTMLInputElement>(null);
+
+	const backgroundUpdate = (): void => {
+		if (isEditing === false) {
+			setValue(String(node.value));
+		}
+	};
+	React.useEffect(backgroundUpdate, [node.value, isEditing]);
+
+	const onFocus = React.useCallback(() => {
+		if (textAreaRef.current !== null) {
+			textAreaRef.current?.select();
+		}
+
+		setIsEditing(true);
+	}, []);
+
+	const onChange = React.useCallback(
+		(_ev: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
+			setValue(data.value);
+		},
+		[],
+	);
+
+	const onBlur = React.useCallback(() => {
+		setValue(String(node.value));
+		setIsEditing(false);
+	}, [node.value]);
+
+	const onKeyDown = React.useCallback(
+		(event: React.KeyboardEvent<HTMLInputElement>) => {
+			const key = event.key;
+			function commitChanges(): void {
+				messageRelay.postMessage(
+					SendEditData.createMessage({
+						containerKey,
+						fluidObjectId: node.fluidObjectId,
+						newData: value,
+						editType: [EditType.string],
+					}),
+				);
+			}
+			if (key === "Enter") {
```

Comment:
Could we add a handler for `escape` to cancel the edit? That's how chrome's devtools work.

Reply:
The one I added should work

Reply:
Do we still need to reset `value` back to the DDS's state / set `isEditing` to `false`? This will (I believe) exit out of the editing dialogue, but will not clear out any edits the user had been making.